torch/utils/file_baton.py

```diff
@@ -44,14 +44,15 @@ def wait(self):
         passed to the constructor.
         """
         has_warned = False
+        should_warn = isinstance(self.warn_after_seconds, (int, float))
```

Comment:
We could do `self.warn_after_seconds is not None` instead, so that we don't silently doing nothing when user provides some non-None value.